src/main/java/bio/terra/workspace/service/workspace/model/WorkspaceRequest.java

```diff
@@ -0,0 +1,70 @@
+package bio.terra.workspace.service.workspace.model;
+
+import bio.terra.workspace.service.spendprofile.SpendProfileId;
+import com.google.auto.value.AutoValue;
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * Internal representation of a request to create a workspace.
+ *
+ * <p>While this object is nearly identical to the {@code Workspace} object, there is a semantic
+ * difference: a {@code Workspace} is a fully formed workspace, while a {@code WorkspaceRequest}
+ * only contains fields specified by clients. These objects may diverge in the future if properties
+ * of a Workspace are generated internally.
+ */
+@AutoValue
+public abstract class WorkspaceRequest {
+
+  /** The globally unique identifier of this workspace */
+  public abstract UUID workspaceId();
+
+  /**
+   * An ID used for idempotency. WorkspaceRequests with matching operationIds are considered
+   * duplicate requests.
+   */
+  public abstract String operationId();
+
+  /**
+   * The spend profile ID associated with this project, if one exists.
+   *
+   * <p>In the future, this will correlate to a spend profile in the Spend Profile Manager. For now,
+   * it's just a unique identifier. To associate a GCP project with a workspace, the workspace must
+   * have a spend profile. They are not needed otherwise.
+   */
+  public abstract Optional<SpendProfileId> spendProfileId();
+
+  /** Temporary feature flag indicating whether this workspace uses MC Terra features. */
+  public abstract WorkspaceStage workspaceStage();
+
+  public static WorkspaceRequest.Builder builder() {
+    return new AutoValue_WorkspaceRequest.Builder();
+  }
+
+  @AutoValue.Builder
+  public abstract static class Builder {
+    public abstract WorkspaceRequest.Builder workspaceId(UUID value);
+
+    public abstract WorkspaceRequest.Builder operationId(String value);
+
+    public abstract WorkspaceRequest.Builder spendProfileId(Optional<SpendProfileId> value);
+
+    public abstract WorkspaceRequest.Builder workspaceStage(WorkspaceStage value);
+
+    public abstract WorkspaceRequest build();
+  }
+
+  /**
+   * Convenience method for getting a WorkspaceRequest builder with pre-filled default values.
+   *
+   * <p>This provides default values for operationID (random UUID), spend profile
+   * (Optional.empty()), and workspace stage (RAWLS_WORKSPACE).
+   */
+  public static WorkspaceRequest.Builder defaultRequestBuilder(UUID workspaceId) {
+    return builder()
+        .workspaceId(workspaceId)
+        .operationId(UUID.randomUUID().toString())
+        .spendProfileId(Optional.empty())
+        .workspaceStage(WorkspaceStage.RAWLS_WORKSPACE);
```

Comment:
I don't think this is a good idea to put in the `main` code. I think this is great for a test utility, but makes it more likely we don't fix places where this is created when we add more fields in the main code.
Tests don't care about details unrelated to what they're verifying. Main code should be explicit, unless there really are reasonable defaults. I don't think these are reasonable defaults in our main code.

Reply:
sg, moved this into WorkspaceServiceTest.